### DISCUSSION TITLE
Distributed inference with JAX: GPU/TPU interconnect

### DISCUSSION THREAD

**Discussion by jwang194** [jax-ml/jax#23216] [Q&A]
We’re interested in acquiring a machine with several GPUs in order to begin some benchmarking of JAX/Tensorflow's sharded HMC framework. As part of our choice of machine, we’d like to know if GPU interconnect topology can significantly reduce overhead. For example, is the jax.psum() backbone of the sharded log density built on a Reduce operation, and can it therefore leverage GPU-GPU direct communication via e.g. NVLink? 

More generally, is there reason to believe that the current implementation of sharded HMC can take advantage of a toroidal or fully connected GPU network?
